src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -130,54 +123,105 @@ int IVsEditorFactory.Close()
-                    {
-                        designerLoader.Initialize(_oleServiceProvider, vsHierarchy, (int)itemid, (IVsTextLines)textBuffer);
-                        pbstrEditorCaption = designerLoader.GetEditorCaption((int)readOnlyStatus);
+                case "Code":
 
-                        var designer = designerService.CreateDesigner(_oleServiceProvider, designerLoader);
-                        ppunkDocView = Marshal.GetIUnknownForObject(designer.View);
-                        pguidCmdUI = designer.CommandGuid;
-                    }
-                    catch
+                    (pbstrEditorCaption, pguidCmdUI, ppunkDocView) = GetCodeEditorData(textBuffer, readOnlyStatus, initializeView: true);
+                    break;
+
+                default:
+
+                    return VSConstants.E_INVALIDARG;
+            }
+
+            ppunkDocData = Marshal.GetIUnknownForObject(textBuffer);
+
+            return VSConstants.S_OK;
+        }
+
+        public object GetDocumentData(uint grfCreate, string pszMkDocument, IVsHierarchy pHier, uint itemid)
+        {
+            var editorAdaptersFactoryService = _componentModel.GetService<IVsEditorAdaptersFactoryService>();
+
+            var contentTypeRegistryService = _componentModel.GetService<IContentTypeRegistryService>();
+            var contentType = contentTypeRegistryService.GetContentType(ContentTypeName);
+            IVsTextBuffer? textBuffer = editorAdaptersFactoryService.CreateVsTextBufferAdapter(_oleServiceProvider, contentType);
+
+            if (_encoding)
+            {
+                if (textBuffer is IVsUserData userData)
+                {
+                    // The editor shims require that the boxed value when setting the PromptOnLoad flag is a uint
+                    var hresult = userData.SetData(
+                        VSConstants.VsTextBufferUserDataGuid.VsBufferEncodingPromptOnLoad_guid,
+                        (uint)__PROMPTONLOADFLAGS.codepagePrompt);
+
+                    if (ErrorHandler.Failed(hresult))
                     {
-                        designerLoader.Dispose();
-                        throw;
+                        Marshal.ThrowExceptionForHR(hresult);
                     }
+                }
+            }
 
-                    break;
+            return textBuffer;
+        }
 
-                case "Code":
+        public object GetDocumentView(uint grfCreate, string pszPhysicalView, IVsHierarchy pHier, IntPtr punkDocData, uint itemid)
+        {
+            throw new NotImplementedException();
+        }
+
+        public string GetEditorCaption(string pszMkDocument, string pszPhysicalView, IVsHierarchy pHier, IntPtr punkDocData, out Guid pguidCmdUI)
+        {
+            var physicalView = pszPhysicalView ?? CodePhysicalViewName;
+            var readOnlyStatus = READONLYSTATUS.ROSTATUS_NotReadOnly;
+
+            IVsTextBuffer? textBuffer = null;
+            if (punkDocData != IntPtr.Zero)
```

Comment:
It feels a bit odd we're mutating the buffer here as a part of GetEditorCaption; does this need to be here or should it be in one of the other methods?

Reply:
Which line are you referring to?
`textBuffer `and `punkDocData `are not being mutated in this method.